app/models/concerns/fog_extensions/vsphere/server.rb

```diff
@@ -40,7 +40,7 @@ def scsi_controller_type
       def select_nic(fog_nics, nic)
         nic_attrs = nic.compute_attributes
         all_networks = service.list_networks(datacenter: datacenter)
-        vm_network = all_networks.detect { |network| nic_attrs['network'] && [network[:name], network[:id], network[:_ref]].compact.include?(nic_attrs['network']) }
+        vm_network = all_networks.detect { |network| nic_attrs['network'] && [network[:name], network[:id]].compact.include?(nic_attrs['network']) }
         vm_network ||= all_networks.detect { |network| network[:_ref] == nic_attrs['network'] }
```

Comment:
I wouldn't mind droping the `_ref` even here, but could we at least move the name here as well, so we know the `id` is the primary attribute?
```ruby
vm_network = all_networks.detect { |network| network[:id] == nic_attrs['network'] }
vm_network ||= all_networks.detect { |network| nic_attrs['network'] && [network[:name], network[:_ref]].compact.include?(nic_attrs['network']) }
```
Or even drop the `_ref` completely and make both lines simple.

Reply:
I don't feel like I understand this code well enough to make a judgement on the proposed change nor to test it in any way that would confirm if either way is better.  The root cause I can find is that some of the objects in our vCenter have _ref's that match other object's id's.  Any fix which ensures a positive match on id is always selected in preference to any possible match against _ref should fix the issue.

Reply:
Understood, this change make sense as is.
I will propose the cleanup in followup PR, if you are uncomfortable to do it ;)